api/app/controllers/application_controller.rb

```diff
@@ -1,2 +1,51 @@
 class ApplicationController < ActionController::API
+    # Prevent CSRF attacks by raising an exception.
+    # For APIs, you may want to use :null_session instead.
+    #protect_from_forgery with: :null_session
+
+    respond_to :json
+
+    include ActionController::HttpAuthentication::Token::ControllerMethods
+
+    before_action :configure_permitted_parameters, if: :devise_controller?
+    before_action :authenticate_user
+
+    def underscore_params!
+        params.deep_transform_keys!(&:underscore)
+    end
+
+
+    private
+
+    #added academic/graduation year
+    def configure_permitted_parameters
+        devise_parameter_sanitizer.permit(:sign_up, keys: [:username, :email, :password, :academic_year, :graduation_year])
+        devise_parameter_sanitizer.permit(:account_update, keys: [:email, :password, :password_confirmation, :current_password])
+    end
+
+    def authenticate_user
+        if request.headers['Authorization'].present?
+          authenticate_or_request_with_http_token do |token|
+            begin
+              jwt_payload = JWT.decode(token, Rails.application.secrets.secret_key_base).first
+    
+              @current_user_id = jwt_payload['id']
+            rescue JWT::ExpiredSignature, JWT::VerificationError, JWT::DecodeError
+              head :unauthorized
+            end
+          end
+        end
+    end
+
+    def authenticate_user!(options = {})
+        head :unauthorized unless signed_in?
+    end
+
+    def current_user
```

Comment:
Just thought it might be nice to add a comment here since this line is so critical for the controller actions.